### PR TITLE
PHP 8.3 support added

### PR DESCRIPTION
**Pull request by glo71317**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by samsonasik** [comment]
laminas-file should no longer be supported imo

**Comment by froschdesign** [comment]
@samsonasik 
When someone provides a pull request and their help, we merge it - just like before.

**Comment by glo71317** [comment]
@froschdesign Can you take a look the PR if all looks fine to you please proceed for merging?

File: phpcs.xml [hunk 1]
```diff
@@ -1,9 +1,198 @@
...
+    <rule ref="SlevomatCodingStandard.TypeHints.ParameterTypeHint.UselessAnnotation">
+        <exclude-pattern>src/Transfer/Adapter/AbstractAdapter.php</exclude-pattern>
+    </rule>
+    <rule ref="WebimpressCodingStandard.Formatting.RedundantParentheses.MultipleExpression">
+        <exclude-pattern>src/Transfer/Adapter/AbstractAdapter.php</exclude-pattern>
+        <exclude-pattern>src/ClassFileLocator.php</exclude-pattern>
+    </rule>
+    <rule ref="WebimpressCodingStandard.Strings.NoConcatenationAtTheEnd.ConcatenationAtTheEnd">
+        <exclude-pattern>src/Transfer/Adapter/AbstractAdapter.php</exclude-pattern>
+    </rule>
```
Comment (Ocramius) [inline phpcs.xml]: Unsure about approach this: if `laminas/laminas-file` will never have CS improved, then we may as well drop the dependency instead.

Perhaps @laminas/technical-steering-committee has more ideas

Comment (glo71317) [reply phpcs.xml]: Yes we can plan for drop but not now we are having the direct dependency in our Magento so slowly we have to see and analyse the uses of this library at our end after this planned release.
I think for now you people can release this if your side there is no any dependency on this.

Comment (Ocramius) [reply phpcs.xml]: @glo71317 I'm aware that you work closely with Adobe Commerce on this, but the discussion here is just internal and on whether `laminas/laminas-file` should still use `squizlabs/php_codesniffer`.

Comment (glo71317) [reply phpcs.xml]: @Ocramius Thanks for clarify! Can you please tell us to whom we can discuss on whether `laminas/laminas-file `should still use `squizlabs/php_codesniffer` or not.

Comment (Ocramius) [reply phpcs.xml]: I already pinged the relevant stakeholders here.

**Comment by weierophinney** [comment]
My recommendation is to not do any CS changes in this PR, and we can open a new PR to do CS changes later.

If the proposed changes fail the CS rules already present in this PR, then they need to follow and use the existing rules present here to keep it focused.

**Comment by gsteel** [comment]
@glo71317 
If you can avoid upgrading laminas-coding-standard in _this_ patch - i.e. if ~1.0.0 is installable on 8.1 through 8.3, then reverting those changes would be the easiest course of action.

If this is not possible, we'll need to upgrade laminas-coding-standard to ~2.5.0 in a separate patch and _fix_ all of the coding standard violations _before_ attempting to add support for PHP 8.3

**Comment by Ocramius** [comment]
@gsteel what about removing the dependency, and re-introducing it in a new PR later?

**Comment by gsteel** [comment]
@Ocramius - I guess removing the dep here would also work, but as @weierophinney suggested that the patch should pass existing CS rules, it would make sense to just keep with version 1.0 here - CS wasn't failing - it's the schema validation that's failing - CS was running fine on 8.0 as far as I could tell, so it's less work to just stay on 1.0 right?

**Comment by glo71317** [comment]
@Ocramius @gsteel I have updated laminas-coding-standard:"~1.0.0" instead of "~2.5.0", Now PHPCodeSniffer checks are failing. earlier was also same failure then i decided to update laminas-coding-standard:"~2.5.0".  So, Can you please suggest?

**Comment by glo71317** [comment]
@Ocramius @gsteel As you suggested to avoid to suppress the CS Changes, I have updated now Can you please guide us how to proceed on this PR?

**Comment by gsteel** [comment]
@Ocramius and @weierophinney 

I've just looked into this - it's not possible to run `phpcs@2.7` on PHP 8.0 anyway, leave alone 8.1.

Last year, almost to the day https://github.com/laminas/laminas-file/pull/13 was merged without CS running either.

Given that tests are passing, do you think we can merge this as-is, and defer CS tooling upgrades for another year, or perhaps discuss continued maintenance of this component in the next TSC meet?

**Comment by froschdesign** [comment]
@localheinz 
> Given that tests are passing, do you think we can merge this as-is, and defer CS tooling upgrades for another year, or perhaps discuss continued maintenance of this component in the next TSC meet?

You are a maintainer for this package; what are your thoughts?

**Review by Ocramius — APPROVED** [review]
Even with red CI CS checks, I'm happy with this, as CS is not as relevant here, IMO.

Will squash-merge, as a warning to PR authors.

Thanks @glo71317!

**Comment by localheinz** [comment]
@froschdesign 

Apologies for the delay, I’m traveling in Canada right now and only have my phone with me.

**Pull request merged**